src/python/pants/util/strutil.py

```diff
@@ -133,3 +133,17 @@ def strip_prefix(string: str, prefix: str) -> str:
         return string[len(prefix) :]
     else:
         return string
+
+
+# NB: We allow bytes because `ProcessResult.std{err,out}` uses bytes.
+def strip_v2_chroot_path(v: Union[bytes, str]) -> str:
```

Comment:
Thoughts on making this a boolean on `Process`, and then doing it for local executions here: https://github.com/pantsbuild/pants/blob/c60024498ed4615d01d78b19ee1ffab7656d34ff/src/rust/engine/process_execution/src/local.rs#L438-L439 ? You would be able to directly remove the `workdir_path`.

Reply:
Worth considering, but it feels wrong to me to leak this to Rust. I appreciate that Rust simply captures the stderr/out and returns it unmodified. Rust transforming that output feels wrong. 

I don’t feel too strongly about this, though. Let me know if you think this is the way to go and I can make the change.

Reply:
Yea, I'm not really sure either. The leak when this is done in Python is that this function hardcodes a guess about where processes will be executed.

Reply:
Agreed. We'll have a leak either way. 

I think it's best to have that leak in Python and make it as contained as possible - for example, do not make this a method on `ProcessResult`. Make it a bit more involved for call sites to do this hacky thing of rewriting the tool's output.

The best fix is to fix the underlying tools themselves; the next best is to isolate our hacky fix to a very small surface area.